frontend/app/helpers/spree/structured_data_helper.rb

```diff
@@ -17,15 +17,16 @@ def structured_product_hash(product)
         {
           '@context': 'https://schema.org/',
           '@type': 'Product',
-          '@id': "#{spree.root_url}product_#{product.id}",
           url: spree.product_url(product),
           name: product.name,
           image: structured_images(product),
           description: product.description,
-          sku: product.sku,
+          sku: structured_sku(product),
+          mpn: structured_sku(product),
           offers: {
             '@type': 'Offer',
-            price: product.price_in(current_currency).amount,
+            price: product.default_variant.price,
```

Comment:
Good point, I'll fix that.

On a side note, would there be any objection to adding a Barcode field at the product and variants level, like the SKU?

With barcodes often unique between variants it would be a handy default field to have built in.

If this wouldn't added on a PR then no problem, I'll just fix the currency price issue as above, but if this is a desirable option I can add the barcode fields at product and variants level, and work that directly into the structured data as a fresh PR?

Let me know what you think?

Reply:
@MatthewKennedy let's keep this PR small and add Barcode in another PR. Barcode would make it to Spree 4.2 and this PR can be still applied to 4.1.x

Reply:
No problem, I'll get this done now.

Reply:
See PR#10122 going to close this.